static/js/popovers.js

```diff
@@ -476,6 +477,8 @@ export function toggle_actions_popover(element, id) {
             editability_menu_item = $t({defaultMessage: "View source"});
         }
 
+        const should_display_mark_as_unread = !message.unread;
```

Comment:
To work around the bug with not having the API this requires, I changed this to only offer the option if the `fetch_status` suggests it'll do what the user expects. I'm hoping we'll get to revise this back to the original in a week or so.